dask/array/core.py

```diff
@@ -531,43 +569,73 @@ def dask_slice(out_name, in_name, shape, blockdims, indexes,
     Example
     -------
 
-    >>> dask_slice('y', 'x', (100,), [(20, 20, 20, 20, 20)], (slice(10, 35),))  #  doctest: +SKIP
+    >>> fancy_slice('y', 'x', (100,), [(20, 20, 20, 20, 20)], (slice(10, 35),))  #  doctest: +SKIP
     {('y', 0): (getitem, ('x', 0), (slice(10, 20),)),
      ('y', 1): (getitem, ('x', 1), (slice( 0, 15),))}
+
+    See Also
+    --------
+
+    take - handle slicing with lists ("fancy" indexing)
+    dask_slice - handle slicing with slices and integers
+    """
+    where_list = [i for i, ind in enumerate(indexes) if isinstance(ind, list)]
```

Comment:
do you really want to explicitly restrict array indexing to lists?

Assuming numpy is a hard dep of dask (which I think it is?) I would rather cast to ndarray for non integer/slices and then allow only 1d arrays of integers. For large arrays, using lists is going to be a bottleneck.